### DISCUSSION TITLE
Understanding Yul

### DISCUSSION THREAD

**Discussion by Not-Sarthak** [Cyfrin/foundry-full-course-cu#2421] [Q&A] [answered]
I am currently on the Upgradable contracts lesson and trying to understand proxies and low level Yul but having a hard time getting a grasp over it. What are some good ways to practise it and understand it in a better way?
A good blog would also help.
Thanks!

**Comment by EngrPips** [comment, marked as answer]
Hello @Not-Sarthak, I think you should try to build a smart contract that is upgradeable all by yourself, and that will help drive the learning home. This is also a good blog https://blog.chain.link/upgradable-smart-contracts/

**Reply by Not-Sarthak** [reply]
will have a read, thanks! I tried writing the contract myself but a bit hard to understand the logic while writing one.

**Reply by EngrPips** [reply]
That is normal if this is your first contact with an Upgradeable smart contract. With time, you will get more comfortable with it.
